packages/flutter/test/services/platform_network_test.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // This test ensures the zone variable used in Dart SDK does not change.
+  //
+  // If this symbol changes, then update [allowHttp] function as well.
+  test('Zone variable can override HTTP behavior', () async {
+    final HttpClient httpClient = HttpClient();
+    expect(() => runZoned(
```

Comment:
I did not for two reasons:

- It requires adding a http server other than localhost which does not work on all machines (I actually tried).

- `allowHttp` implementation is tested in Dart SDK so it would be a duplicate of these tests:

https://github.com/dart-lang/sdk/blob/master/tests/standalone_2/io/http_ban_http_embedder_test.dart
https://github.com/dart-lang/sdk/blob/master/tests/standalone_2/io/http_ban_http_normal_test.dart

Reply:
will they produce a different error message with or without the allowHttp with not http server?
It does not seem this test can be useful without checking the allowHttp

Reply:
I think I know where you were getting at.

- I added a test to make sure that `allowHttp` sets the right zone variable.
- I also made sure the StateError being thrown is the right state error. Unfortunately, the only way to do that is to look at the exception string as there's no error code. This should be very stable though (since it is an internal message) so I don't have concerns.

Please take a look.

Reply:
Unless you have other ideas, this is likely the best I can do without setting up a working HTTP server. As I mentioned, I would not want to duplicate the SDK tests here.